src/services/decoders/ClpIrDecoder/utils.ts

```diff
@@ -0,0 +1,48 @@
+import {MainModule} from "clp-ffi-js";
+
+
+enum CLP_IR_STREAM_TYPE {
+    STRUCTURED = "structured",
+    UNSTRUCTURED = "unstructured",
+}
+
+/**
+ * Keys that delineate the auto-generated and user-generated namespaces in structured IR log events.
+ */
+interface StructuredIrNamespaceKeys {
+    autoGenerated: string;
+    userGenerated: string;
+}
+
+/**
+ * Retrieves the structured IR namespace keys from the "clp-ffi-js" module.
+ *
+ * @param ffiModule The main module from "clp-ffi-js".
+ * @return
+ * @throws {Error} If the keys are of invalid type.
+ */
+const getStructuredIrNamespaceKeys = (ffiModule: MainModule): StructuredIrNamespaceKeys => {
```

Comment:
I believe we can get rid of these type assertions if we upgrade emscripten to the latest version in the clp-ffi-js repo. for now, let's track the issue at https://github.com/y-scope/clp-ffi-js/issues/69